### DISCUSSION TITLE
Missing box collisions (Collision Query Demo)

### DISCUSSION THREAD

**Discussion by Cray-1** [bepu/bepuphysics2#219] [Q&A] [answered]
Hi, I added a bunch of statics and box queries to the Collision Query Demo and discovered that sometimes the queries did not result in a collision when they should. It seems only to happen when I apply a transformation to the queries. The simplest example I could reproduce is to remove the mesh in the CollisionQueryDemo.cs (line 46) and instead of the 5x5 queries (lines 228-234) adding only this query: `queries.Allocate(BufferPool) = new Query { Box = new Box(2, 2, 2), Pose = new RigidPose(new Vector3(0, 0, 3), Quaternion.CreateFromAxisAngle(new Vector3(1, 0, 0), MathF.PI * -0.5f)) };`
This query box clearly intersects the static ground plane box but does not turn green. When removing the Quaternion rotation it does turn green. I would be happy for any help.

**Comment by RossNordby** [comment, marked as answer]
Oops! The orientations passed to the batcher were swapped, so it interpreted the query's orientation as the static's orientation. Thanks for the report, the demo's fixed in 66971de28a60f84d281cb3198eb973f9a9682f0d.

**Reply by Cray-1** [reply]
Thanks for the quick fix!
